### DISCUSSION TITLE
How to recreate a CLI convert statement  magick-wasm

### DISCUSSION THREAD

**Discussion by netbizltd** [dlemstra/magick-wasm#85] [Q&A]
I have this convert statement on the CLI:

```
$MAGICK convert "image.jpg" \
  -rotate 180 \
  -resize "600%" \
  -virtual-pixel HorizontalTile -background black +distort Polar 0 +repage \
  -resize "1000x1000" \
  -gravity center -background black -extent "1200x1200" \
  -quality 95 "result.jpg"
```

How would I go about writing this in JS?
**Thanks in advance!**

Note: The order of things is important here: Rotate 180° needs to be done first so the polar distortion goes in the correct direction. The purpose of resize to 600%, then polar distort and then scale down to 1000x1000 is for anti-aliasing purposes (since the polar distort makes ugly pixelated images). The gravity center on a bigger canvas is for making black borders.

**Comment by dlemstra** [comment]
It looks like everything is available in the Api but I want to make sure you get the same output image. Can you share your input image so I can double check this?

**Comment by netbizltd** [comment]
Hi Dirk,

After lots more trial-and-error I think I got it!

Here's a test image:

![image](https://github.com/dlemstra/magick-wasm/assets/97948726/0ce67288-e586-4174-84a2-81bffecbc9f6)

and here's what I have now: 
```
ImageMagick.read(file, image => {
  image.virtualPixelMethod = VirtualPixelMethod.HorizontalTile
  image.backgroundColor = MagickColors.Black
  image.quality = 95
  image.rotate(180)
  image.resize(new MagickGeometry('600%'))
  image.distort(DistortMethod.Polar, [0])
  image.compositeGravity(image, Gravity.Center)
  image.extent(new MagickGeometry('1:1'), Gravity.Center)
  image.resize(POLARSIZE, POLARSIZE)
  image.extent(RESULTSIZE, RESULTSIZE, Gravity.Center)
  image.write(.....
})
```

which gets me to this result:

![image](https://github.com/dlemstra/magick-wasm/assets/97948726/461ad183-81f5-4cdb-b4f7-33d65b16180a)

Thank you so much for all your effort!
Cheers, Chris

**Reply by dlemstra** [reply]
Thanks for sharing. Here are a couple comments on your code:

```js
ImageMagick.read(file, image => {
  image.virtualPixelMethod = VirtualPixelMethod.HorizontalTile
  // You can also specify this in one of the `extent` overloads
  image.backgroundColor = MagickColors.Black
  image.quality = 95
  image.rotate(180)
  image.resize(new MagickGeometry('600%'))
  image.distort(DistortMethod.Polar, [0])
  image.compositeGravity(image, Gravity.Center) // I don't think you need this.
  image.extent(new MagickGeometry('1:1'), Gravity.Center)
  image.resize(POLARSIZE, POLARSIZE)
  image.extent(RESULTSIZE, RESULTSIZE, Gravity.Center)
  image.write(.....
})
```

**Reply by netbizltd** [reply]
The `image.backgroundColor = MagickColors.Black` is for the polar distortion (otherwise circle on white).

Yes, the `image.compositeGravity(image, Gravity.Center)` was a leftover - not nescessary.

Cheers
